myTinyURL/convert/views.py

```diff
@@ -1,29 +1,31 @@
-from django.shortcuts import render, render_to_response, get_object_or_404
-from django.http import HttpResponse, HttpResponseRedirect
+from django.shortcuts import render, get_object_or_404
+from django.http import HttpResponse, HttpResponseRedirect, HttpRequest, JsonResponse
 from .models import URL
 from django.conf import settings
-import random, string
-
+import random, string, json
 # Create your views here.
 
 def index(request):
-    return HttpResponse("Hello, welcome to my shorturl convert page.")
+    return render(request, 'convert/index.html')
 
-def shorten(request, url):
+def shorten(request):
+    url = request.POST['url']
+    print("received short url request:{}".format(url))
     if (url == ''):
-        return HttpResponse("Error...Empty url")
-    print 
+        print("received url is empty")
+        return JsonResponse("Error: blank url...", safe=False)
     short_id = gen()
```

Comment:
We need a core engine to support two functions 
1. string short2long(string)
2. string long2short(string)

Put this core engine into a separate file with a new class. So that Views.py calls the core engine.